addons/sale/models/sale.py

```diff
@@ -577,6 +577,18 @@ def _compute_qty_delivered_updateable(self):
         for line in self:
             line.qty_delivered_updateable = (line.order_id.state == 'sale') and (line.product_id.track_service == 'manual') and (line.product_id.expense_policy == 'no')
 
+    @api.depends('invoice_lines.invoice_id.state', 'invoice_lines.invoice_id.refund_invoice_id.state')
```

Comment:
Dependencies need to depend on the parents too, in order to work properly.
So for invoice_lines.invoice_id.state you need to have invoice_lines, invoice_lines.invoice_id and invoice_lines.invoice_id.state.

But I'm not certain doing this through a compute method is the best way...